src/main/js/dygraph-light/main/LabelMaker.ts

```diff
@@ -200,25 +199,26 @@ class Metadata {
 	}
 }
 
-const getFinalLabel = (linking: 'overlap' | 'concatenate', requestedLabels: string[], lblIdx: number, object: MetaWithTableFormat, colName: string, postfix: string = '') => {
-	if (linking === 'concatenate'){
-		return (requestedLabels.length > lblIdx && requestedLabels[lblIdx].length
-			? requestedLabels[lblIdx]
-			: getLabel(object.tableFormat, colName)) + postfix;
+const getFinalLabel = (linking: 'overlap' | 'concatenate', requestedLabel: string, object: MetaWithTableFormat, colName: string, postfix: string = '') => {
+	if (requestedLabel) return requestedLabel + postfix;
 
-	} else {
-		const filename = object.filename;
+	if (linking === 'concatenate'){
+		return getLabel(object.tableFormat, colName) + postfix;
+	} 
+	const filename = object.filename;
 
-		return (requestedLabels.length > lblIdx && requestedLabels[lblIdx].length
-			? requestedLabels[lblIdx]
-			: filename.slice(0, filename.lastIndexOf('.'))) + postfix;
-	}
+	return filename.slice(0, filename.lastIndexOf('.')) + postfix;
 };
 
-const getLegendLabelsFromParams = (params: UrlSearchParams, name: string) => {
-	return params.has(name)
-		? params.get(name).split(',') as string[]
-		: [];
+const getLegendLabelFromParams = (params: UrlSearchParams, name: string, objId: string) => {
+	const labels: string[] = params.has(name) ? params.get(name).split(',') : [];
```

Comment:
Not sure if it's better, but I believe this could be written as
```suggestion
	const labels: string[] = params.get(name)?.split(',') || [];
```
The main benefit, in my view, is that this operates on the actual value we get, rather than the non-null expectation of that value based on the result of another check.